lib/ex_aws/sns.ex

```diff
@@ -169,6 +169,77 @@ defmodule ExAws.SNS do
     request("ListSubscriptions", %{"NextToken" => next_token})
   end
 
+  ## Messages
+  ######################
+
+  @notification_params ["Message", "MessageId", "Subject", "Timestamp", "TopicArn", "Type"]
+  @optional_notification_params ["Subject"]
+  @confirmation_params ["Message", "MessageId", "SubscribeURL", "Timestamp", "Token", "TopicArn", "Type"]
+  @signature_params ["SignatureVersion", "Signature", "SigningCertURL"]
+  @message_types ["SubscriptionConfirmation", "UnsubscribeConfirmation", "Notification"]
+
+  @doc "Verify message signature"
+  def verify_message(message_params) do
+    with :ok <- validate_message_params(message_params),
+      :ok <- validate_signature_version(message_params["SignatureVersion"]),
+      {:ok, public_key} <- ExAws.SNS.PublicKeyCache.get(message_params["SigningCertURL"]) do
+        message_params
+        |> get_string_to_sign
+        |> verify(message_params["Signature"], public_key)
+    else
+      {:error, error} -> {false, error}
```

Comment:
Not really a fan of `{false, error}`. What's wrong with `{:ok, msg} | {:error, reason}` as a return signature?

On that note, we need typespecs.